### ISSUE TITLE
Release kernel sources for Xiaomi Mi A1

### ISSUE THREAD

**Issue by ravinder0003** [MiCode/Xiaomi_Kernel_OpenSource#765]
Please release kernel sources for Mi A1 asap.
Thanks

**Comment by Aniketh-Raj** [comment]
Hello Xiaomi, Please release the Kernel Sources for Mi A1 TISSOT.. The phone was released some time back and still Kernel Sources are not up even when the device is being called as Android ONE...!! Please release the sources asap

**Comment by subhangjoshi16** [comment]
Give M1 a1 sources even if it is Android one device then also no source not released till now

**Comment by kotzir** [comment]
Release Mi A1 kernel sources please

**Comment by ves626** [comment]
Xiaomi, please _release Mi A1 kernel_ ASAP, devs are still waiting...

**Comment by ashishranjan738** [comment]
Xiaomi please the release the kernel sources ASAP

**Comment by kunalshah912** [comment]
Come ON Xiaomi ! You are not as bad as OnePlus is "with handling Customers!" Release the Kernel source for Mi A1....

**Comment by rickling1993** [comment]
Please release Mi A1 kernel sources. It is a shame that an Android One device by Google partnership didn't follow GPL requirement.

**Comment by DamandeepS** [comment]
Please provide the source code.

**Comment by kotzir** [comment]
I contacted Xiaomi Global Support and they replied that they are gonna forward my request to the appropriate department of Xiaomi. Just waiting, for a couple of time for now.

**Comment by ghost** [comment]
release the source code.

**Comment by luk1337** [comment]
no~

**Comment by marines** [comment]
@luk1337

**Comment by luca020400** [comment]
@marines

**Comment by TechExhibeo** [comment]
hakuna your tatas

**Comment by luca020400** [comment]
Swahili on github, that's new.

**Comment by methuselahin** [comment]
Please release kernel sources for this device. I bought this thinking this is google android one device and source code will be available as soon as this phone is launched after few days.  Even one of elite developers of LINEAGE is here I can see that which makes me happy.

**Comment by TechExhibeo** [comment]
^ Sure thing, till evening it should be resolved :D
And thanks for reminding :)

**Comment by percy-g2** [comment]
Please provide the source code.

**Comment by Aymahalawa** [comment]
Pls release the Mi A1 kernel source , This is Android one phone !!

**Comment by jaikrishna835** [comment]
https://www.change.org/p/people-xiomi-mi-a1-kernel-source?recruiter=677404613&utm_source=share_petition&utm_medium=copylink&utm_campaign=share_petition&utm_term=share_for_starters_page

**Comment by jaikrishna835** [comment]
Sign the above petition friends

**Comment by bennyhillthebest** [comment]
Please release the source code Xiaomi. The Mi A1 is the nexus successor

**Comment by luk1337** [comment]
successor? I think you have no idea what you're talking about.

**Comment by hugwalk** [comment]
大家都等很久了(
Source 就快给吧…
Devs 都没办法开发

**Comment by linpapa** [comment]
hi, please release the sourse code, it is no harm

**Comment by HenrikNyberg** [comment]
You have released Kernel source to other phones even in the redmi series, this is a AndroidOne, what is taking so long, please respect GPL2! This is my first Xiaomi please don't make it my last..

**Comment by luca020400** [comment]
FYI: theoretically, google should handle android one kernel releases.

**Comment by abler98** [comment]
I already really want to start developing under this device, how long to wait for the kernel source?

**Comment by methuselahin** [comment]
XIAOMI common what is taking so long. I am now regretting purchasing this crap device seriously. Your own ROM is bugged with so many issues. Why isn't google listening to these comments? HTC and MOTOROLA are way better at handling things when it comes to developers. You proved again and again that why I forced people in the past not to buy your devices. There is no respect for GPL. You proved me right. I am regretting that I Buy your device.

**Comment by ghost** [comment]
Release Please.

**Comment by asrupesh** [comment]
Hello Xiaomi, Please release the Kernel Sources for Mi A1.. The phone was released more than 2 months and still Kernel Sources are not up even when the device is being called as Android ONE...!! Please release the sources ASAP.......

**Comment by adeveloper86** [comment]
Hello Xiaomi, Please release the Kernel Sources for Mi A1..

**Comment by shahidcodes** [comment]
What's taking too much time? One plus released kernel source and device for 5T within days. Much disappointed

**Comment by DeicPro** [comment]
You are requited by GPLv2 license to release source code, just do it!

**Comment by scafroglia93** [comment]
CIAO MAMMA

**Comment by luca020400** [comment]
SONO IN TV!1!!1!1!11

**Comment by percy-g2** [comment]
![](https://thumbs.gfycat.com/RepulsiveUnitedHarvestmouse-size_restricted.gif)

**Comment by zeroxxx2** [comment]
Bunch of crying babies. Be patient or GTFO.

**Comment by methuselahin** [comment]
At least now wake up 
https://www.xda-developers.com/xiaomi-not-released-kernel-sources-mi-a1/

**Comment by Kizoky** [comment]
@zeroxxx2 Sure mate. I'm still waiting for Redmi 4 prada source code to be released, and this is still not done after 1 year.

**Comment by nilmarsh555** [comment]
Shame on you Xiaomi. Kernel source code for one plus 5x released on first day of launch.

**Comment by mohanen** [comment]
Still no Xiaomi_Kernel_OpenSource Maintainers have replied to this thread . Just Great

**Comment by acnalogia** [comment]
Xiaomi, please release Mi A1 kernel ASAP

**Comment by scafroglia93** [comment]
I don't understand, you know that Xiaomi behaves this way and you continue to buy it

**Comment by methuselahin** [comment]
@scafroglia93 see even now they are doing this. I mean this company never respected GPL in the past too what can be expected now?

**Comment by shahidcodes** [comment]
No fucking kernel source yet.. its been 3 months to this issue creation. How much time more?

**Comment by luk1337** [comment]
As much as they like :^)

**Comment by mohanen** [comment]
https://www.gnu.org/licenses/gpl-violation.html
license-violation@gnu.org

**Comment by zeroxxx2** [comment]
Yeah right, when it comes to a violation if it's really punished almost nobody would be able to produce Android devices anymore. Nobody gets hurt even if they don't release it.

Suck it up, wait for them or move on.

**Comment by mohanen** [comment]
@zeroxxx2  dude nobody asked for your advice, move on

**Comment by zeroxxx2** [comment]
Dude I can be anywhere, do you pay for my internet connection? Suck it up and move on.

**Comment by ashu00** [comment]
https://github.com/cuteprince/android_kernel_xiaomi_msm8953

**Comment by ishantvivek** [comment]
I think, he ported kernel ^ based on kumajaya guide, like he did for Redmi 2.

**Comment by ashu00** [comment]
^ Yes, he is porting from redmi note 4.

**Comment by KonradIT** [comment]
You're violating the GPL, if you sell the Mi A1 in the US you can get sued.

**Comment by dong-dev** [comment]
15 days before the end of year, still waiting

**Comment by luk1337** [comment]
cool

**Comment by ishantvivek** [comment]
yup very cool weather here^ , how do you come to know? 🤔🤔😂😂

**Comment by shahidcodes** [comment]
So cool that my hands are freezing..

**Comment by deyjishnu** [comment]
Please release the kernel source

**Comment by scafroglia93** [comment]
Merry Christmas

**Comment by dong-dev** [comment]
##  Good News: Xiaomi just release Android Oreo Today (GMT+7)  
##  Other News: maybe 3 month is over.

**Comment by Beneter** [comment]
I will buy a Mi A1 immediately after you releasing the sources. But not before!

**Comment by FoxBob79** [comment]
> I will buy a Mi A1 immediately after you releasing the sources. But not before!
  
I make the mistake to buy the Mi A1 bevor the release of the sources. I hope that would not be a great mistake!!

**Comment by subashsn** [comment]
I've got this bad feeling that they have backdoored it, and won't ever be releasing the source :/

**Comment by luk1337** [comment]
Do you really think you'd be getting same source code as stock ROM? Take a look at other devices, they all got a special source code releases for the public audience lol.

**Comment by anetar** [comment]
xiaomi has had a long-standing issue with kernel sources. The Linux kernel’s GPLv2 license requires kernel sources to be released in a timely manner,
top 1 oneplus released  kernel sources

**Comment by scafroglia93** [comment]
@Beneter LOL

Because you haven't see BQ Github

**Comment by dvignesh1995** [comment]
@subashsn omg that might just be true lol
maybe they did backdoor or something like that and decided never to release the kernel source

**Comment by subashsn** [comment]
@luk1337 Didn't realize that. Figured that one cannot be sure there is no backdoor unless

1. every bit of the hardware, software of a smartphone is open sourced
2. they are able to verify what they have matches the source
3. they are smart enough to analyze the source to say there is no backdoor

Which makes my worry pointless, so please release the source so that awesome people can make  custom ROMs with ease XD

**Comment by adeveloper86** [comment]
Release Kernel source to start development

**Comment by scafroglia93** [comment]
Of course you're funny, buying Xiaomi devices you know what you're going to and yet you keep complaining constantly. 
Next time don't buy if you have source problems, look for a manufacturer with a more ethical profile

**Comment by ghost** [comment]
Good news. Xiaomi released kernel source of Mi A1 😃

https://github.com/MiCode/Xiaomi_Kernel_OpenSource/tree/tissot-n-oss

**Comment by scafroglia93** [comment]
Source is completely outdated based on 3.18.31 but now MI A1 use 3.18.66 so i think this is for Nougat

Well done Xiaomi

**Comment by abler98** [comment]
@scafroglia93 Look at the name of the branch - tissot-**n**-oss

**Comment by luk1337** [comment]
Eh, you don't need to cry about it. N release is enough for real devs out there.

**Comment by scafroglia93** [comment]
They are useless sources as if they were released for Android 4.4 when we are now at version 8.1

@luk1337 
In fact i don't buy this Chinese rubbish just for the cost

**Comment by luk1337** [comment]
No, they aren't useless. I say this as a person that got only M kernel source while there was N already released.

**Comment by luk1337** [comment]
@scafroglia93 
>It's difficult for an OEM upload the entire source without maxi kang squashed commit ?

Ask Samsung, LG, HTC, ... they don't give you full .git as well.

**Comment by alex9099** [comment]
So... no oreo sources?
